packages/common/src/converter/payload-converter.ts

```diff
@@ -140,3 +149,75 @@ export class BinaryPayloadConverter extends AsyncFacadePayloadConverter {
     return content.data as any;
   }
 }
+
+/**
+ * Converts between protobufjs Message instances and serialized Protobuf Payload
+ */
+export class ProtobufPayloadConverter extends AsyncFacadePayloadConverter {
+  public encodingType = encodingTypes.METADATA_ENCODING_PROTOBUF;
+
+  constructor(protected protobufClasses?: Record<string, unknown>) {
+    super();
+
+    if (protobufClasses && typeof protobufClasses !== 'object') {
+      throw new DataConverterError('protobufClasses must be an object');
+    }
+  }
+
+  public toDataSync(value: unknown): Payload | undefined {
```

Comment:
Are you changing the API to async in the next PR?

Reply:
Yep